pkg/cloudprovider/aws/instancetype.go

```diff
@@ -319,19 +312,65 @@ func (i *InstanceType) kubeReservedResources(pods int64) v1.ResourceList {
 			resources[v1.ResourceCPU] = *cpuOverhead
 		}
 	}
+	if kc != nil && kc.KubeReserved != nil {
+		return lo.Assign(resources, kc.KubeReserved)
+	}
 	return resources
 }
 
-func (i *InstanceType) miscResources(vmMemOverhead float64) v1.ResourceList {
+func (i *InstanceType) evictionThreshold(kc *v1alpha5.KubeletConfiguration, vmMemoryOverhead resource.Quantity) v1.ResourceList {
+	overhead := v1.ResourceList{
+		v1.ResourceMemory: resource.MustParse("100Mi"),
+	}
+	if kc != nil && kc.EvictionHard != nil {
+		if v, ok := kc.EvictionHard[memoryAvailable]; ok {
+			if strings.HasSuffix(v, "%") {
+				p, err := strconv.ParseFloat(strings.Trim(v, "%"), 64)
+				if err != nil {
+					panic(fmt.Sprintf("expected percentage value to be a float but got %s, %v", v, err))
+				}
+				// Setting percentage value to 100% is considered disabling the threshold according to
+				// https://kubernetes.io/docs/reference/config-api/kubelet-config.v1beta1/
+				if p == 100 {
+					p = 0
+				}
+				// Calculation is node.capacity * evictionHard[memory.available] if percentage
+				// From https://kubernetes.io/docs/concepts/scheduling-eviction/node-pressure-eviction/#eviction-signals
+				totalAllocatable := i.resources.Memory().DeepCopy()
+				totalAllocatable.Sub(vmMemoryOverhead)
+				overhead[v1.ResourceMemory] = resource.MustParse(fmt.Sprint(math.Ceil(float64(totalAllocatable.Value()) / 100 * p)))
+			} else {
+				overhead[v1.ResourceMemory] = resource.MustParse(v)
+			}
+		}
+	}
+	return overhead
+}
+
+func (i *InstanceType) miscResources(overheadPercentage float64) v1.ResourceList {
 	memory := i.memory().Value()
 	return v1.ResourceList{
-		v1.ResourceMemory: resource.MustParse(fmt.Sprintf("%dMi",
-			// vm-overhead
-			(int64(math.Ceil(float64(memory)*vmMemOverhead/1024/1024)))+
-				// eviction threshold https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/pkg/kubelet/apis/config/v1beta1/defaults_linux.go#L23
-				100,
-		)),
+		v1.ResourceMemory: resource.MustParse(fmt.Sprintf("%dMi", int64(math.Ceil(float64(memory)*overheadPercentage/1024/1024)))),
+	}
+}
+
+func (i *InstanceType) computeMaxPods(ctx context.Context, kc *v1alpha5.KubeletConfiguration) *int64 {
+	amiFamily := amifamily.GetAMIFamily(i.provider.AMIFamily, &amifamily.Options{})
+	var mp *int64
+
+	switch {
+	case kc != nil && kc.MaxPods != nil:
+		mp = ptr.Int64(int64(ptr.Int32Value(kc.MaxPods)))
+	case !injection.GetOptions(ctx).AWSENILimitedPodDensity:
+		mp = ptr.Int64(110)
+	default:
+		mp = ptr.Int64(i.eniLimitedPods())
+	}
+
+	if kc != nil && ptr.Int32Value(kc.PodsPerCore) > 0 && amiFamily.PodsPerCoreEnabled() {
+		mp = ptr.Int64(lo.Min([]int64{int64(ptr.Int32Value(kc.PodsPerCore)) * ptr.Int64Value(i.VCpuInfo.DefaultVCpus), ptr.Int64Value(mp)}))
```

Comment:
This shouldn't use `DefaultVCpus` since those are virtual CPUs and the kubelet uses physical cores. You can use `DefaultCores` instead.  

https://github.com/kubernetes/kubernetes/blob/442574f3a7321ff0f34dec9bae67f28651dbabd2/pkg/kubelet/nodestatus/setters.go#L282

https://github.com/kubernetes/kubernetes/blob/55e1d2f9a73a0c78fc0c266236c3e31261635b90/pkg/kubelet/cm/cpumanager/topology/topology.go#L221

Reply:
This actually isn't the case. Weirdly enough, kubelet is using `DefaultVCPus` from the instance when testing with instances that have a differing number of cpus and cores.